git_externals/git_externals.py

```diff
@@ -306,18 +306,28 @@ def git_update_checkout(reset):
         else:
             echo('Checking out branch', git_externals[ext_repo]['branch'])
             egit('checkout', git_externals[ext_repo]['branch'])
-            egit('pull', 'origin', git_externals[ext_repo]['branch'])
 
-            if git_externals[ext_repo]['ref'] is not None:
-                echo('Checking out commit', git_externals[ext_repo]['ref'])
-                ref = git_externals[ext_repo]['ref']
-                if git_externals[ext_repo]['ref'].startswith('svn:'):
-                    ref = egit('log', '--grep', 'git-svn-id:.*@%s' % ref.strip('svn:r'), '--format=%H').strip()
-                egit('checkout', ref)
+            rev = get_rev(ext_repo)
+            if rev is not None:
+                echo('Checking out commit', rev)
+                egit('checkout', rev)
+
+    def get_rev(ext_repo, mode='git'):
+        assert mode in ('git', 'svn'), "mode = {} not in (git, svn)".format(mode)
```

Comment:
`auto` is supposed not to outlive the normalization, here we must know the vcs kind.